Source/AliveLibAO/Bullet.cpp

```diff
@@ -35,4 +43,202 @@ BaseGameObject* Bullet::VDestructor(signed int flags)
+                SFX_Play_43AE60(SoundEffect::AirStream_28, field_2C_scale != FP_FromDouble(0.5) ? 90 : 60, 2000, 0);
+                SFX_Play_43AE60(SoundEffect::MeatBounce_43, field_2C_scale != FP_FromDouble(0.5) ? 90 : 60, Math_RandomRange_450F20(300, 700), 0);
+                SFX_Play_43AE60(SoundEffect::KillEffect_64, field_2C_scale != FP_FromDouble(0.5) ? 90 : 60, Math_RandomRange_450F20(900, 1400), 0);
+            }
+            field_6_flags.Set(BaseGameObject::eDead_Bit3);
+        }
+        case BulletType::Type_2:
+        {
+            FP v5;
+            FP v7;
+            FP hitX;
+            FP hitY;
+            PSX_RECT shootRect = {};
+            if (sControlledCharacter_50767C == (BaseAliveGameObject *) gElum_507680)
+            {
+                v5 = gElum_507680->field_B4_velx * FP_FromInteger(4);
+                v7 = sActiveHero_507678->field_A8_xpos;
+            }
+            else
+            {
+                v5 = sActiveHero_507678->field_B4_velx * FP_FromInteger(2);
+                if (field_18_xpos >= sActiveHero_507678->field_A8_xpos)
+                {
+                    v7 = sActiveHero_507678->field_A8_xpos + FP_FromInteger(field_34 << 4);
+                }
+                else
+                {
+                    v7 = sActiveHero_507678->field_A8_xpos - FP_FromInteger(field_34 << 4);
+                }
+            }
+            shootRect.x = FP_GetExponent(v7 - v5);
+            shootRect.w = shootRect.x + 2;
+            shootRect.y = FP_GetExponent(sActiveHero_507678->field_AC_ypos)
+                + sActiveHero_507678->field_10_anim.Get_FrameHeader_403A00(-1)->field_8_data.points[2].y //or points 3?!
+                - 10;
+            shootRect.h = shootRect.y + 10;
+            BaseAliveGameObject *pShotObj = ShootObject_409400(&shootRect);
+            if (pShotObj->VTakeDamage_401920(this))
+            {
+                SFX_Play_43AE60(SoundEffect::AirStream_28, 90, 2000, 0);
+                SFX_Play_43AE60(SoundEffect::MeatBounce_43, 90, Math_RandomRange_450F20(300, 700), 0);
+                SFX_Play_43AE60(SoundEffect::KillEffect_64, 90, Math_RandomRange_450F20(900, 1400), 0);
+                field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                return;
+            }
+            if (sCollisions_DArray_504C6C->RayCast_40C410(
+                field_18_xpos,
+                field_1C_ypos,
+                v7 - v5,
+                sActiveHero_507678->field_AC_ypos + FP_FromInteger(10),
+                &field_14_pLine,
+                &hitX,
+                &hitY,
+                0x400) == 1)
+            {
+                auto pSpark = ao_new<Spark>();
+                if (pSpark)
+                {
+                    pSpark->ctor_477B70(hitX, hitY, FP_FromInteger(1), 9u, -31, 159);
+                }
+                New_Particles_419A80(hitX, hitY, FP_FromInteger(1), 3, 0);
+            }
+            if (Math_RandomRange_450F20(0, 128) < 64)
```

Comment:
could be an if/else to play the sfx and put the  field_6_flags.Set(BaseGameObject::eDead_Bit3); after it?